### PR TITLE
[spinel] fix unused variables

### PR DESCRIPTION
**Pull request by wgtdkp**

### CODE REVIEW

File: src/lib/spinel/radio_spinel_impl.hpp [hunk 1]
```diff
@@ -2150,6 +2150,8 @@ uint32_t RadioSpinel<InterfaceType, ProcessContextType>::GetBusSpeed(void) const
 template <typename InterfaceType, typename ProcessContextType>
 void RadioSpinel<InterfaceType, ProcessContextType>::HandleRcpUnexpectedReset(spinel_status_t aStatus)
 {
+    OT_UNUSED_VARIABLE(aStatus);
```
Comment (jwhui) [inline src/lib/spinel/radio_spinel_impl.hpp:2153]: What if the log level is set to `NONE`?

Comment (wgtdkp) [reply src/lib/spinel/radio_spinel_impl.hpp:2153]: I think this change is required when log level is `NONE`.

**Review by jwhui — APPROVED** [review]

**Comment by jwhui** [comment]
@wgtdkp , can you help rebase?

**Comment by wgtdkp** [comment]
> @wgtdkp , can you help rebase?

Done.

**Pull request merged**